tv_apps/browser/js/settings.js

```diff
@@ -1,15 +1,667 @@
+    var end_event = (function() {
+      Settings.settingsDialogHomepageOk.removeEventListener('transitionend',
+          end_event, false);
+      Settings.hideDialogHomepage();
+    });
+    // end animantion event
+    Settings.settingsDialogHomepageOk.addEventListener('transitionend',
+        end_event, false);
+    document.activeElement.classList.remove('active');
+    this.setHomepage(this.settingsDialogHomepageInputArea.value);
+    this.settingsHomepageName.textContent =
+      this.settingsDialogHomepageInputArea.value;
+  },
+
+  /**
+   * Handle search engine click.
+   */
+  handleSearchEngineClick: function settings_handleSearchEngineClick() {
+    if( !this.isDialogSearchDisplayed() ) {
+      this.focusList = new Array();
+      Browser.switchCursorMode(false);
+      this.settingsDialog.classList.remove('hidden');
+      this.settingsDialogSearch.classList.remove('hidden');
+
+      this.selectSearchEngine = this.currentSearchEngine;
+      this.searchEngineList = SearchUtil.getEngineList();
+      for(var i in this.searchEngineList) {
+        var title = document.createElement('div');
+        title.classList.add('title');
+        var name = document.createElement('p');
+        name.classList.add('name');
+        name.textContent = this.searchEngineList[i].name;
+        title.appendChild(name);
+
+        var check = document.createElement('div');
+        check.classList.add('check');
+        var onoff = document.createElement('p');
+        if( name.textContent == this.currentSearchEngine ) {
+          name.dataset.display = 'selected';
+          onoff.classList.add('on');
+        } else {
+          name.dataset.display = '';
+          onoff.classList.add('off');
+        }
+        check.appendChild(onoff);
+
+        var item = document.createElement('div');
+        item.classList.add('item');
+        item.appendChild(title);
+        item.appendChild(check);
+        item.tabIndex = '0';
+        this.settingsDialogSearchSelectList.appendChild(item);
+        item.addEventListener('mouseup',
+          this.handleDialogSearchSelected.bind(this));
+        item.addEventListener('keyup',
+          this.handleDialogSearchSelected.bind(this));
+        this.focusList.push(item);
+      }
+      this.focusList.push(this.settingsDialogSearchOk);
+      this.focusList.push(this.settingsDialogSearchCancel);
+      this.focusPos = this.focusList.length - 1;
+      this.focusChange(this.focusPos);
+    }
+  },
+
+  handleClearHistoryClick: function settings_handleClearHistoryClick() {
+    BrowserDialog.createDialog('clear_history', null);
+  },
+
+  handleClearCookieClick: function settings_handleClearCookieClick() {
+    BrowserDialog.createDialog('del_cookie', null);
+  },
+
+  clearCookieFailed: function settings_clearCookieFailed() {
+    // this.settingsBannerMessage.innerHTML = 'Failed to clear cookies and stored data';
+    this.settingsBannerMessage.innerHTML = _('LT_FAILED_');
```

Comment:
this should be:

`this.settingsBannerMessage.setAttribute('data-l10n-id', 'LT_FAILED_');